src/compiler/CMakeLists.txt

```diff
@@ -79,9 +79,9 @@ add_library(acpp-clang SHARED
 
 target_include_directories(acpp-clang PRIVATE
   ../../include
-  ${LLVM_INCLUDE_DIRS}
   ${CMAKE_CURRENT_SOURCE_DIR}
   ${CMAKE_BINARY_DIR}/include)
+target_include_directories(acpp-clang SYSTEM PRIVATE ${LLVM_INCLUDE_DIRS})
```

Comment:
Yes I added it to make cmake add `-isystem` to suppress warnings in LLVM headers. I remember there were some issues with the order of some system headers, but I thought this was related to how/where AdaptiveCpp was installed, see #1170 and the issues linked there. 

I don't think that this will cause issues here, I thought this was standard practice to suppress warnings from library headers.